grype/match/ignore.go

```diff
@@ -215,7 +215,11 @@ func ifPackageLocationApplies(location string) ignoreCondition {
 func ifUpstreamPackageNameApplies(name string) ignoreCondition {
 	return func(match Match) bool {
 		for _, upstream := range match.Package.Upstreams {
-			if name == upstream.Name {
+			pattern, err := packageNameRegex(name)
+			if err != nil {
+				continue
+			}
```

Comment:
I think this should be moved outside of the returned func, the same way `ifPackageNameApplies` checks the regex once and returns an "always false" func e.g.:

```
func ifUpstreamPackageNameApplies(name string) ignoreCondition {
	pattern, err := packageNameRegex(name)
	if err != nil {
		return func(Match) bool { return false }
	}
	return func(match Match) bool {
		for _, upstream := range match.Package.Upstreams {
			if pattern.MatchString(upstream.Name) {
				return true
			}
		}
		return false
	}
}